framework/transport/ieee1905_transport/ieee1905_transport_packet_processing.cpp

```diff
@@ -21,6 +22,44 @@ namespace mapf {
 
 static const uint8_t ieee1905_max_message_version = 0x00;
 
+void Ieee1905Transport::update_neighbours(const Packet &packet)
+{
+    if (packet.ether_type != ETH_P_1905_1) {
+        MAPF_DBG("Ignoring non-1905 packet");
+        return;
+    }
+    if (packet.src_if_type != CmduRxMessage::IF_TYPE_NET) {
+        MAPF_DBG("Ignoring packet from src_if_type " << int(packet.src_if_type));
+        return;
+    }
+
+    // Delete aged neighbors
```

Comment:
This shouldn't happen for every received packet.
Can be performed every 30/35 seconds (half of the aging duration).
Otherwise, large number of neighbours may have negative impact on packet processing time.

Reply:
I disagree. We are doing the same worst case `O(n)` search 2 lines below in order to update the `last_seen` or add a new neighbor, so if we end up not deleting neighbors, its a linear complexity.

Most importantly - this is an EasyMesh home network. The number of neighbors is actually the number of other repeaters connected directly to this multiap agent, and this should be a very small number - mininum one neighbor, which is the backhaul, normal cascaded will be 2.
Much more than that doesn't make any sense. 

Bottom line - I don't think packet processing time will be impacted at all, and doing this as it is now is much simpler and easier to maintain.

Reply:
There's a different between a search with a worst case of `O(n)` and iterating over the entire list which is a constant `O(n)`.
I guess the number of neighbors depends on the architecture of the mesh deployment.
Nothing prevents one from deploying a network in a star topology where all (or most) repeaters are connected to the gateway over a wired backhaul. Since we don't enforce any limitations, theoretically we can end up with thousands of neighbors.
But, since I'm not sure if that's a real life scenario, I won't insist on this one :)